homeassistant/components/switchbot_cloud/vacuum.py

```diff
@@ -0,0 +1,135 @@
+# "Dormant": "Dormant",
+# "InTrouble": "In trouble",
+# "InRemoteControl": "In remote control",
+# "InDustCollecting": "In dust collecting"
+
+# fan speed
+# "0": "Quiet",
+# "1": "Standard",
+# "2": "Strong",
+# "3": "MAX"
+VACUUM_SWITCHBOT_STATE_TO_HA_STATE: dict[str, str] = {
+    "StandBy": STATE_IDLE,
+    "Clearing": STATE_CLEANING,
+    "Paused": STATE_PAUSED,
+    "GotoChargeBase": STATE_RETURNING,
+    "Charging": STATE_DOCKED,
+    "ChargeDone": STATE_DOCKED,
+    "Dormant": STATE_IDLE,
+    "InTrouble": STATE_ERROR,
+    "InRemoteControl": STATE_CLEANING,
+    "InDustCollecting": STATE_DOCKED,
+}
+
+
+# https://github.com/OpenWonderLabs/SwitchBotAPI?tab=readme-ov-file#robot-vacuum-cleaner-s1-plus-1
+class SwitchBotCloudVacuum(SwitchBotCloudEntity, StateVacuumEntity):
+    """Representation of a SwitchBot vacuum."""
+
+    _attr_supported_features: VacuumEntityFeature = (
+        VacuumEntityFeature.BATTERY
+        | VacuumEntityFeature.FAN_SPEED
+        | VacuumEntityFeature.PAUSE
+        | VacuumEntityFeature.RETURN_HOME
+        | VacuumEntityFeature.START
+        | VacuumEntityFeature.STATE
+    )
+
+    _attr_name = None
+    _attr_fan_speed_list: list[str] = ["quiet", "standard", "strong", "max"]
+
+    async def async_set_fan_speed(self, fan_speed: str, **kwargs: Any) -> None:
+        """Set fan speed."""
+        if fan_speed in self._attr_fan_speed_list:
+            self._attr_fan_speed = fan_speed
+            await self.send_api_command(
+                VacuumCommands.POW_LEVEL,
+                parameters=str(self._attr_fan_speed_list.index(fan_speed)),
+            )
+            self.async_write_ha_state()
+
+    async def async_pause(self) -> None:
+        """Pause the cleaning task."""
+        await self.send_api_command(VacuumCommands.STOP)
+
+    async def async_return_to_base(self, **kwargs: Any) -> None:
+        """Set the vacuum cleaner to return to the dock."""
+        await self.send_api_command(VacuumCommands.DOCK)
+
+    async def async_start(self) -> None:
+        """Start or resume the cleaning task."""
+        await self.send_api_command(VacuumCommands.START)
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        if not self.coordinator.data:
+            return
+
+        # 'deviceId' | 'deviceMac': 'XXXXX'
+        # 'deviceType': 'K10+' | 'WoSweeperMini'
+        # 'workingStatus': 'ChargeDone' | 'Charging' | 'GotoChargeBase' | 'Clearing' | 'Paused' | ...
+        # 'onlineStatus': 'online'
+        # 'battery': 100
+        self._attr_battery_level = self.coordinator.data.get("battery")
+        self._attr_available = self.coordinator.data.get("onlineStatus") == "online"
+
+        switchbot_state = self.coordinator.data.get("workingStatus")
+        if switchbot_state in VACUUM_SWITCHBOT_STATE_TO_HA_STATE:
+            self._attr_state = VACUUM_SWITCHBOT_STATE_TO_HA_STATE[switchbot_state]
```

Comment:
```suggestion
        self._attr_state = VACUUM_SWITCHBOT_STATE_TO_HA_STATE.get(switchbot_state)
```
This way if there's a state not in the list, it will show up as unknown